wikimedia/wikipedia-ios
# Talk page cache

**Phabricator:** https://phabricator.wikimedia.org/T315330

### Notes
* This PR creates caching for the new talk pages, using the `SharedContainerCache` to persist .json files. It also adds a housekeeping task to remove older cached pages.

### Test Steps
1. Go to a talk page
2. leave the page, turn off the internet connection and return to the same page. It should load the cached topics
3. To test the housekeeping, export your cached data to your computer. Go To Settings >Help and Feedback > Export User data. Check how many items you have under Talk Page Cache. 
4. Lower the max cached items on `deleteStaleCachedItems` so you don't have to save 50 pages 
5. Follow the steps described here (test step 7) https://github.com/wikimedia/wikipedia-ios/pull/4211

Execute the background database cleanup task. Set a breakpoint right after task scheduling [here](https://github.com/wikimedia/wikipedia-ios/blob/b2a693672852457a39763681820f7f3abd761b97/Wikipedia/Code/AppDelegate.m#L263). Background the app, then when you hit the breakpoint print this in the console:

> e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateLaunchForTaskWithIdentifier:@"org.wikimedia.wikipedia.databaseHousekeeper"]

After a little bit you should see a printout in the console saying:

> Simulating launch for task with identifier org.wikimedia.wikipedia.databaseHousekeeper

When you see this, hit the play button in the debugger to continue executing the program. The task will run and then the console will print out:

> Marking simulated task complete: <BGProcessingTask: org.wikimedia.wikipedia.databaseHousekeeper>

6. Now export your data again, and check if the older entries were deleted
